jerry-core/lit/lit-magic-strings.h

```diff
@@ -77,6 +78,19 @@ typedef enum
   LIT_MAGIC_STRING__COUNT /**< number of magic strings */
 } lit_magic_string_id_t;
 
+/**
+ * the [[RegExpMatcher]] internal slot handling in ECMA v6 and v5.1
+ */
+#if ENABLED (JERRY_ES2015)
+#define LIT_MAGIC_STRING_REGEXP_MATCHER LIT_INTERNAL_MAGIC_STRING_REGEXP_MATCHER
+#else /* !ENABLED (JERRY_ES2015) */
+#if ENABLED (JERRY_BUILTIN_REGEXP)
+#define LIT_MAGIC_STRING_REGEXP_MATCHER LIT_MAGIC_STRING_REGEXP_UL
+#else /*!ENABLED (JERRY_BUILTIN_REGEXP) */
+#define LIT_MAGIC_STRING_REGEXP_MATCHER LIT_MAGIC_STRING_OBJECT_UL
```

Comment:
Why do we need to define this as `LIT_MAGIC_STRING_OBJECT_UL`? Cases with this magic string should not be reachable if the regexp builtins are disabled.